lib/core/service/auth/auth_service.dart

```diff
@@ -4,28 +4,33 @@ import 'package:app/core/domain/auth/entities/auth_session.dart';
 import 'package:app/core/domain/user/entities/user.dart';
 import 'package:app/core/failure.dart';
 import 'package:app/core/oauth/oauth.dart';
+import 'package:app/core/service/firebase/firebase_service.dart';
 import 'package:app/injection/register_module.dart';
 import 'package:dartz/dartz.dart';
 import 'package:injectable/injectable.dart';
 
 @LazySingleton()
 class AuthService {
   final appOAuth = getIt<AppOauth>();
+  final firebaseService = getIt<FirebaseService>();
   Stream<OAuthTokenState> get tokenStateStream => appOAuth.tokenStateStream;
 
   Future<Either<Failure, bool>> login() async {
     var res = await appOAuth.login();
     return res.fold(
       (l) => Left(Failure()),
       (success) {
-        if (success) return const Right(true);
-
+        if (success) {
+          firebaseService.addFcmToken();
```

Comment:
I recommend not putting fcmToken logic inside auth service, 
In firebase service, let create a listener on `getIt<AppOAuth>().tokenStateStream, 
When ever the tokenState changed to valid, then we should addFcmToken to our server

Reply:
Yep, let me adjust